music21/noteworthy/translate.py

```diff
@@ -453,20 +486,39 @@ def translateNote(self, attributes):
         durationInfo = attributes['Dur']
         pitchInfo = attributes['Pos']
 
-        n = note.Note()   # note!
+        p = self.getPitchFromPositionInfo(pitchInfo)
+
+        if self.currentClef == 'PERCUSSION':
+            n = note.Unpitched()
+            n.fromDisplayPitch(p)
+        else:
+            n = note.Note()   # note!
+            n.pitch = p
```

Comment:
this should be:

`n = note.Note(p)`

will speed up creation substantially.  (Otherwise a default C4 pitch is created and then dropped).